crates/flox/src/commands/channel.rs

```diff
@@ -76,122 +88,131 @@ pub struct Search {
-                    if first {
-                        first = false;
-                        print!("[ ");
-                    } else {
-                        print!(", ");
-                    }
-                    println!("{}", line);
-                }
-                println!("]");
-            } else {
-                for line in stdout.lines() {
-                    let entry: serde_json::Value = serde_json::from_str(line).unwrap();
-                    let input = entry.get("input").unwrap().as_str();
-                    let pname = entry.get("pname").unwrap().as_str();
-                    let attr_path = entry.get("path").unwrap().as_array().unwrap();
-                    let mut path: Vec<String> = Vec::new();
-                    for part in attr_path {
-                        path.push(part.as_str().unwrap().to_string())
-                    }
-                    print!("{}#{}: {}", input.unwrap(), path.join("."), pname.unwrap());
-
-                    if let Some(ver) = entry.get("version").unwrap().as_str() {
-                        print!("@{}", ver)
-                    }
-
-                    if let Some(desc) = entry.get("description").unwrap().as_str() {
-                        print!(" - {}", desc)
-                    }
-                    println!();
-                }
-            }
+            // FIXME: We may have warnings on `stderr` even with a successful call to `pkgdb`.
+            //        We aren't checking that at all at the moment because better overall error handling
+            //        is coming in a later PR.
+            let search_results = SearchResults::try_from(output.stdout.as_slice())?;
+            render_search_results(search_results, self.json)?;
             Ok(())
         } else {
+            let err_msg = String::from_utf8_lossy(&output.stdout);
             bail!(
-                "pkgdb exited with status code {}",
-                output.status.code().unwrap_or(-1)
+                "pkgdb exited with status code {}: {}",
+                output.status.code().unwrap_or(-1),
+                err_msg
             );
         }
     }
 }
 
+fn construct_search_params(search_term: &Option<String>, flox: &Flox) -> Result<SearchParams> {
+    let channels = flox
+        .channels
+        .iter()
+        .filter_map(|entry| {
+            if HIDDEN_CHANNELS.contains_key(&*entry.from.id) {
+                None
+            } else if DEFAULT_CHANNELS.contains_key(&*entry.from.id) {
+                Some((ChannelType::Flox, entry))
+            } else {
+                Some((ChannelType::User, entry))
+            }
+        })
+        .sorted();
+
+    // Create `registry` parameter for `pkgdb`
+    let mut priority: Vec<String> = Vec::new();
+    let mut inputs = HashMap::new();
+    for (_, entry) in channels {
+        priority.push(entry.from.id.to_string());
+        let input = RegistryInput {
+            from: FlakeRef::Indirect(entry.from.clone()),
```

Comment:
Why not pass a direct ref?

Reply:
What's in `Flox` is already an `IndirectRef` so constructing the `FlakeRef::Indirect` variant is trivial. All we need this for is the ability to serialize to a string anyway.

Reply:
I mean
```suggestion
            from: entry.to.clone(),
```
That avoids having to do a registry lookup in the future

Reply:
Fixed